starknet-core/src/types/typed_data/encoder.rs

```diff
@@ -342,8 +342,12 @@ impl Encoder {
                     }
                 };
 
-                get_selector_from_name(str_value)
-                    .map_err(|_| TypedDataError::InvalidSelector(str_value.to_owned()))?
+                match str_value.strip_prefix("0x") {
```

Comment:
It appears that parsing a selector regardless of prefix (existing behavior) is the expected behavior given that the type is literally `Selector`.

IIUC this change here is to make `starknet-rs` deviate from SNIP-12 to be compatible with the de-facto standard `starknet.js`? If so could you please annoate the same way as other such deviations in the codebase?